tests/unit/testsuite/Bolt/Boltpay/Block/Checkout/BoltpayTest.php

```diff
@@ -255,6 +289,401 @@ public function testGetSaveOrderURL()
+     *
+     * @param string|null $multiStepKey             the multi-step publishable key, empty string or null indicates not set
+     * @param string|null $paymentOnlyKey           the payment-only publishable key, empty string or null indicates not set
+     * @param string      $routeName                The magento path route name
+     * @param string      $controllerName           The magento path controller name
+     *
+     * @throws Exception if there is a failure retrieving the focus mock's Request object
+     */
+    private function setUp_getPublishableKeyForThisPage($multiStepKey, $paymentOnlyKey, $routeName, $controllerName) {
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Block_Checkout_Boltpay')
+            ->setMethods(array('getPublishableKey'))
+            ->disableOriginalConstructor()
+            ->disableOriginalClone()
+            ->disableArgumentCloning()
+            ->getMock()
+        ;
+        $this->currentMock->getRequest()
+            ->setRouteName($routeName)
+            ->setControllerName($controllerName);
+
+        $mockProxy = new Bolt_Boltpay_Block_Checkout_Boltpay();
+
+        $this->app->getStore()->setConfig('payment/boltpay/publishable_key_multipage', $multiStepKey);
+        $this->app->getStore()->setConfig('payment/boltpay/publishable_key_onepage', $paymentOnlyKey);
+
+        $this->currentMock->expects($this->exactly(2))->method('getPublishableKey')
+            ->withConsecutive(
+                [$this->equalTo('multi-page')],
+                [$this->equalTo('one-page')]
+            )
+            ->willReturnOnConsecutiveCalls(
+                $mockProxy->getPublishableKey('multi-page'),
+                $mockProxy->getPublishableKey('one-page')
+            )
+        ;
+    }
+
+    /**
+     * @test
+     * that the appropriate product key is returned for a particular page when at least a multi-step key or payment only
+     * key has been configured
+     *
+     * @covers ::getPublishableKeyForThisPage
+     * @dataProvider getPublishableKeyForThisPage_withAtLeastOneKeyConfiguredProvider
+     *
+     * @param string|null $multiStepKey             the multi-step publishable key, empty string or null indicates not set
+     * @param string|null $paymentOnlyKey           the payment-only publishable key, empty string or null indicates not set
+     * @param string      $routeName                The magento path route name
+     * @param string      $controllerName           The magento path controller name
+     * @param string      $expectedReturnValue      The key that is expected to be returned by the mock
+     *
+     * @throws Exception if there is a problem in setting up test dependencies
+     */
+    public function getPublishableKeyForThisPage_withAtLeastOneKeyConfigured($multiStepKey, $paymentOnlyKey, $routeName, $controllerName, $expectedReturnValue)
```

Comment:
It's better to use one variable< for example $case as array. 
The original method has no such parameters.

Reply:
Actually, I prefer the style which I am using.  It is more clear in the test and in the documentation, especially with being able to declare types.  The signature of the original method is not relevant here.

Reply:
will not change